src/dex/uniswap-v3/uniswap-v3.ts

```diff
@@ -769,19 +771,59 @@ export class UniswapV3
           const balanceDestToken =
             _destAddress === pool.token0 ? state.balance0 : state.balance1;
 
+          const iterationCount: Record<string, number> = {};
           const unitResult = this._getOutputs(
             state,
             [unitAmount],
             zeroForOne,
             side,
             balanceDestToken,
+            iterationCount,
           );
+          const getOutputsStart = Date.now();
           const pricesResult = this._getOutputs(
             state,
             _amounts,
             zeroForOne,
             side,
             balanceDestToken,
+            iterationCount,
+          );
+          const getOutputsFinish = Date.now();
+
+          const poolKey = `${pool.token0}_${pool.token1}_${
+            pool.feeCodeAsString
+          }_${pool.tickSpacing ?? ''}`;
+
+          const iterationStats = Object.entries(iterationCount).reduce(
+            (acc, [key, value], index) => {
+              if (value > acc.max) {
+                acc.max = value;
+              }
+              if (value < acc.min) {
+                acc.min = value;
+              }
+              acc.avg += value;
+
+              if (index === Object.entries(iterationCount).length - 1) {
+                acc.avg = Math.round(
+                  acc.avg / Object.entries(iterationCount).length,
+                );
+              }
+
+              return acc;
+            },
+            { avg: 0, max: 0, min: Infinity },
+          );
+
+          this.logger.info(
+            `_getOutputs_${poolKey}_${reqId}: ${
+              getOutputsFinish - getOutputsStart
+            } ms (src: ${srcToken.address}, dest: ${
```

Comment:
won't logging tokens be redundant since their addresses are included in the pool key?

Reply:
token0 is not always srcToken, and same for destToken

but I agree that it's a bit redundant, so will just leave src/destTokens in poolKey